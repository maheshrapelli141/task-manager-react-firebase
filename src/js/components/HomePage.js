import React from 'react';
import TaskManagement from '../../TaskManagement.jpg';

const HomePage = () => (
  <img src={TaskManagement} alt="Why task management?" width="100%" />
);

export default HomePage;
