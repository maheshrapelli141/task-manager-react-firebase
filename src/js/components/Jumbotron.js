import React from 'react';
import PropTypes from 'prop-types';

const Jumbotron = ({ children }) => (
  <div className="jumbotron jumbotron-fluid">
    <div className="container">{children}</div>
  </div>
);

Jumbotron.propTypes = {
  children: PropTypes.node.isRequired
};

export default Jumbotron;
