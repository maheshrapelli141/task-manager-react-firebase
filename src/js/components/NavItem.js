import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const NavItem = ({ path, label }) => (
  <li className="nav-item">
    <NavLink to={path} activeClassName="active" className="nav-link">
      {label}
    </NavLink>
  </li>
);

NavItem.propTypes = {
  path: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
};

export default NavItem;
