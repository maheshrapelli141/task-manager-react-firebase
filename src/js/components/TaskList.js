import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import TaskItem from './TaskItem';
import {loadTasks} from '../actions/tasks.actions';
import tasksService  from '../services/task.service';
import { toDateString } from '../util/DateUtil';

export class TaskList extends Component {
  constructor(props) {
    super(props);

    this.props.tasks;
  }

  componentDidMount(){
    this.getAllTasks();
  }

  getAllTasks(){
    tasksService.getAllTasks().then(tasks => {
      const data = tasks.docs.map(task => {
        const data = task.data();
        return {
          ...data,
          id: task.id,
          createdDate: toDateString(data.createdDate.seconds * 1000)
        };
      });

      console.log(data.map(tsk => tsk.createdDate));
      this.props.dispatch(loadTasks(data));
    });
  }

  render() {
    const { tasks } = this.props;
    return (
      <section className="row" id="taskList">
        {tasks.length > 0 ? (
          tasks.map((task, index) => (
            <TaskItem key={task.id} {...task} index={index} />
          ))
        ) : (
          <div className="col-12">
            <div className="alert alert-info" role="alert">
              Task list is empty, please add a task!
            </div>
          </div>
        )}
      </section>
    );
  }
}

TaskList.defaultProps = {
  tasks: []
};

TaskList.propTypes = {
  tasks: PropTypes.arrayOf(PropTypes.object)
};

const mapStateToProps = state => ({ tasks: state.tasks });

export default connect(
  mapStateToProps,
  null
)(TaskList);
