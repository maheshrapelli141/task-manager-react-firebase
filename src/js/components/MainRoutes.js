import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';

import HomePage from './HomePage';
import TaskPage from './TaskPage';
import ReportPage from './ReportPage';
import PrivateRoute from './PrivateRoute';

const MainRoutes = () => (
  <Switch>
    <Route exact path="/" component={HomePage} />
    <PrivateRoute path="/task" component={TaskPage} />
    <PrivateRoute path="/report" component={ReportPage} />
    <Redirect to="/" />
  </Switch>
);

export default MainRoutes;
