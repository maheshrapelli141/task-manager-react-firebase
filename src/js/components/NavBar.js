import React from 'react';
import { connect } from 'react-redux';

import NavItem from './NavItem';
import { auth, provider } from '../util/firebase';

const login = () => { auth.signInWithPopup(provider) }
const logout = () => { auth.signOut() }

const NavBar = props => (
  <nav className="navbar navbar-expand-lg navbar-dark bg-primary">
    <a className="navbar-brand" href="#">
      Navbar
    </a>
    <button
      className="navbar-toggler"
      type="button"
      data-toggle="collapse"
      data-target="#navbarContent"
    >
      <span className="navbar-toggler-icon" />
    </button>

    <div className="collapse navbar-collapse" id="navbarContent">
      {props.user
        ?
          <ul className="navbar-nav mr-auto">
            <NavItem path="/task" label="Task" />
            <NavItem path="/report" label="Report" />
            <button type="button" className="btn btn-primary btn-right" onClick={logout}  >Logout</button>
          </ul>
        :
          <ul className="navbar-nav mr-auto">
            <button type="button" className="btn btn-primary btn-right" onClick={login}  >Login</button>
          </ul>
      }

    </div>
  </nav>
);

const mapStateToProps = state => {
  const { user } = state;
  return {
    user
  };
}

export default connect(mapStateToProps)(NavBar);
