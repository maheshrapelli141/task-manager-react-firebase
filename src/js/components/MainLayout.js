import React from 'react';
import { HashRouter } from 'react-router-dom';

import NavBar from './NavBar';
import MainRoutes from './MainRoutes';

const MainLayout = () => (
  <HashRouter hashType="noslash">
    <div className="wrapper">
      <NavBar />
      <main>
        <div className="card">
          <div className="card-body">
            <MainRoutes />
          </div>
        </div>
      </main>
    </div>
  </HashRouter>
);

export default MainLayout;
