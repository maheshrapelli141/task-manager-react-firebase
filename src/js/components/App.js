import React from 'react';
import { connect } from 'react-redux';
import { auth } from '../util/firebase';
import MainLayout from './MainLayout';
import { login } from '../actions/user.actions';

class App extends React.Component {
  constructor(props) {
    super(props);
    
  }

  componentDidMount() {
    auth.onAuthStateChanged((user) => {
      this.props.dispatch(login(user)) 
    });
  }


  render() {
    return (
      <div id="mainContainer">
        <MainLayout />
      </div>);
  }
}
const mapStateToProps = state => {
  const { user } = state;
  return {
    user
  }
};
export default connect(mapStateToProps)(App);
