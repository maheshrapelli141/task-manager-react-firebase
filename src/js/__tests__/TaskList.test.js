import React from 'react';
import { mount } from 'enzyme';

import { TaskList } from '../components/TaskList';

const setup = customProps => {
  const props = {
    tasks: [],
    ...customProps
  };

  const wrapper = mount(<TaskList {...props} />);

  return {
    props,
    wrapper
  };
};

jest.mock('../components/TaskItem', () => {
  const TaskItem = () => <div />;
  return TaskItem;
});

describe('TaskList component', () => {
  it('should render empty state when tasks is empty', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render tasks', () => {
    const { wrapper } = setup({
      tasks: [
        {
          id: '123',
          name: 'Some task',
          category: 'Cat A',
          createDate: '2018-01-01',
          dueDate: '2018-01-01',
          reminderDate: '2018-01-01',
          description: 'lorem ipsum',
          resolved: true
        }
      ]
    });
    expect(wrapper).toMatchSnapshot();
  });
});
