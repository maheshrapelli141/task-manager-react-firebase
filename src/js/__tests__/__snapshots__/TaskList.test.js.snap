// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList component should render empty state when tasks is empty 1`] = `
<TaskList
  tasks={Array []}
>
  <section
    className="row"
    id="taskList"
  >
    <div
      className="col-12"
    >
      <div
        className="alert alert-info"
        role="alert"
      >
        Task list is empty, please add a task!
      </div>
    </div>
  </section>
</TaskList>
`;

exports[`TaskList component should render tasks 1`] = `
<TaskList
  tasks={
    Array [
      Object {
        "category": "Cat A",
        "createDate": "2018-01-01",
        "description": "lorem ipsum",
        "dueDate": "2018-01-01",
        "id": "123",
        "name": "Some task",
        "reminderDate": "2018-01-01",
        "resolved": true,
      },
    ]
  }
>
  <section
    className="row"
    id="taskList"
  >
    <TaskItem
      category="Cat A"
      createDate="2018-01-01"
      description="lorem ipsum"
      dueDate="2018-01-01"
      id="123"
      index={0}
      key="123"
      name="Some task"
      reminderDate="2018-01-01"
      resolved={true}
    >
      <div />
    </TaskItem>
  </section>
</TaskList>
`;
